keps/sig-network/4963-kube-proxy-flowtables-fastpath/README.md

```diff
@@ -0,0 +1,463 @@
+     |__yes_________________fastpath bypass ____________________________|
+
+             Fig.1 Netfilter hooks and flowtable interactions
+```
+
+Enabling the flowtable fastpath requires to use nftables and only needs to create a flowtable on the corresponding network interface and add one rule in the forward chain.
+
+Example configuration:
+
+```
+table inet x {
+        flowtable f {
+                hook ingress priority 0; devices = { eth0, eth1 };
+        }
+        chain y {
+                type filter hook forward priority 0; policy accept;
+                ip protocol tcp flow add @f
+                counter packets 0 bytes 0
+        }
+}
+```
+
+Kube-proxy will allow users to opt-in for Service acceleration, since it is not possible to guarantee that a feature that completely bypass the kernel has unintended consequences in some environments or incompatibility with other networking components.
+
+The opt-in behavior will be based on network interface names and attributes and will use CEL expressions, similar to how we use `--pod-interface-name-prefix` in [KEP-2450](https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/2450-Remove-knowledge-of-pod-cluster-CIDR-from-iptables-rules). The powerful matching expression allow users to deal with more complex environments: nodes with multiple interfaces and only some of them with hardware offload, matches on alias to select only some specific kind of pods, ...
+
+Some example expressions are:
+
+- `interface.name.startsWith(\"eth\")` selects all interfaces with prefix `eth`
+- `interface.type == \"veth\"` selects all interfaces with type `veth`
+- `interface.alias == \"offload\"` selects all interfaces with alias set to `offload`
+
+This expressions can also be concatenated, see https://github.com/google/cel-go for more information.
+
+Kernel bypassing will only be applied once the connection is ESTABLISHED, the rationale behind this is that, once the connection is accelerated, it bypass the network stack, by waiting for the connection to be established we ensure the integration with other network applications (network policies, service meshes, ... ) in the node and we avoid possible performance problems or possible DOS attacks by trying to accelerate every single connection on a node.
+
+### User Stories (Optional)
+
+#### Story 1
+
+I want to use Service traffic acceleration with kube-proxy to improve the performance of my applications.
+
+### Risks and Mitigations
+
+Once the network traffic moves to the datapath it completely bypass the kernel stack, so
+any other network applications that depend on the packets going through the network stack (monitoring per example) we'll not be able to see the connection data. The feature will only
+apply the fast path on established connections, since most of the network applications are statefuls, usually is safe to think that once a connection is established no additional operations are required on it.
```

Comment:
Small typo:
```suggestion
apply the fast path on established connections, since most of the network applications are stateful, usually is safe to think that once a connection is established no additional operations are required on it.
```